plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/split/HudiBackgroundSplitLoader.java

```diff
@@ -74,7 +77,20 @@ public void start()
                             .map(partition -> Futures.submit(() -> loadSplits(partition), executor))
                             .peek(this::hookErrorListener)
                             .collect(Collectors.toList());
-                    Futures.whenAllComplete(futures).run(asyncQueue::finish, directExecutor());
+                    Futures.addCallback(Futures.whenAllComplete(futures).run(asyncQueue::finish, directExecutor()), new FutureCallback<Object>()
```

Comment:
Guava's change that removed `@CanIgnoreReturnValue`: https://github.com/google/guava/commit/a7f6b089d71b853728e0885334f6dafd2e51be25